kernelci/storage.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2020 Collabora Limited
+# Author: Lakshmipathi Ganapathi <lakshmipathi.ganapathi@collabora.com>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+import os
+import requests
+from urllib.parse import urljoin
+from kernelci import shell_cmd
+
+
+def _upload_files(api, token, path, input_files):
+    headers = {
+        'Authorization': token,
+    }
+    data = {
+        'path': path,
+    }
+    files = {
+        'file{}'.format(i): (name, fobj)
+        for i, (name, fobj) in enumerate(input_files.items())
+    }
+    url = urljoin(api, 'upload')
+    resp = requests.post(url, headers=headers, data=data, files=files)
+    resp.raise_for_status()
+    return True
+
+
+def upload_files(api, token, upload_path, input_files):
+    """Upload rootfs to KernelCI backend.
+
+    *api* is the URL of the KernelCI backend API
+    *token* is the backend API token to use
+    *upload_path* is the target on KernelCI backend
+    *input_files* dictionary of input files
+    """
+    _upload_files(api, token, upload_path, input_files)
+    return True
```

Comment:
This function always returns `True` and doesn't really care about the result.

Reply:
`raise_for_status()` call should handle the negative case and should report any exception.

Reply:
Then, there's no need to return anything.  That's what the original `_upload_files()` did.